dspace-server-webapp/src/main/java/org/dspace/app/rest/SubmissionCCLicenseUrlRepository.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.ServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.exception.DSpaceBadRequestException;
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.model.SubmissionCCLicenseUrlRest;
+import org.dspace.app.rest.model.wrapper.SubmissionCCLicenseUrl;
+import org.dspace.app.rest.repository.DSpaceRestRepository;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.core.Context;
+import org.dspace.license.service.CreativeCommonsService;
+import org.dspace.services.RequestService;
+import org.dspace.utils.DSpace;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * This Repository is responsible for handling the CC License URIs.
+ * It only supports a search method
```

Comment:
You may want to check with @artlowel , but because this endpoint _only supports a search method_ it may not be easily "discoverable" to Angular.  See this bug https://jira.lyrasis.org/browse/DS-4495 which was resolved in #2766.

However, this is easily fixed, if we need to make the search path discoverable.  You'd just want to update this class to `implements InitializingBean` and add an `afterPropertiesSet()` that looks similar to this: https://github.com/DSpace/DSpace/blob/master/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClaimedTaskRestRepository.java#L200-L205

Essentially, the goal here is to provide a secondary link directly to the `/search` subpath, so that the HAL Browser knows of it & also so the Angular UI can easily find it / be aware of it.

Reply:
@tdonohue Followed this up with @artlowel & he confirmed your theory, I added a commit to resolve the issue. The endpoint now shows up in the HAL browser.

Reply:
@KevinVdV : Thanks! that looks good now.  I'll merge it immediately